packages/block-library/src/accordion-item/index.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * Server-side rendering of the `core/accordion-item` block.
+ *
+ * @package WordPress
+ * @since 6.6.0
+ *
+ * @param array $attributes The block attributes.
+ * @param string $content The block content.
+ *
+ * @return string Returns the updated markup.
+ */
+function block_core_accordion_item_render( $attributes, $content ) {
+	if ( ! $content ) {
+		return $content;
+	}
+
+	$p         = new WP_HTML_Tag_Processor( $content );
+	$unique_id = wp_unique_id( 'accordion-item-' );
+
+	while ( $p->next_tag() ) {
+		if ( $p->has_class( 'wp-block-accordion-item' ) ) {
+            $p->set_attribute( 'data-wp-context', '{ "id": "' . $unique_id . '" }' );
+			$p->set_attribute( 'data-wp-class--is-open', 'state.isOpen' );
+			if ( $attributes['openByDefault'] ) {
+				$p->set_attribute( 'data-wp-init', 'callbacks.open' );
+			}
+		}
+	}
+
+	$content = $p->get_updated_html();
+	$p       = new WP_HTML_Tag_Processor( $content );
+
+	while ( $p->next_tag() ) {
+		if ( $p->has_class( 'accordion-item__toggle' ) ) {
```

Comment:
I don't think you need to rewind unless the order of appearance of the tags is unknown.

Ultimately, if it doesn't find a tag, there's no need to add the following ones either, since the block will be broken anyway, won't it?

I would just nest the checks in the correct order, something like this:

```php
if ( $p->next_tag( array( 'class_name' => 'wp-block-accordion-group' ) ) ) {
  // Add directives..
  if ( $p->next_tag( array( 'class_name' => 'accordion-item__toggle' ) ) ) {
    // Add directives..
    if ( $p->next_tag( array( 'class_name' => 'wp-block-accordion-content' ) ) ) {
      // Add directives..

      // If everything went well, overwrite the content.
      $content = $p->get_updated_html();
    }
  }
}
```

If the order is unknown, then you can use a bookmark to rewind:

```php
$p->set_bookmark('start');

if ( $p->next_tag( array( 'class_name' => 'wp-block-accordion-group' ) ) ) {
  // Add directives..

  // Rewind
  $p->seek('start');
  if ( $p->next_tag( array( 'class_name' => 'accordion-item__toggle' ) ) ) {
    // Add directives..

    // Rewind
    $p->seek('start');
    if ( $p->next_tag( array( 'class_name' => 'wp-block-accordion-content' ) ) ) {
      // Add directives..

      // If everything went well, overwrite the content.
      $content = $p->get_updated_html();
    }
  }
}
```

Reply:
Solved in https://github.com/WordPress/gutenberg/pull/64119/commits/4c27428623c99430d65d4cf7ade64a233a9dca14.